pkg/server/server.go

```diff
@@ -2,793 +2,38 @@ package server
-	fetcher      applicationv2.Fetcher
-	glAuthClient auth.GitlabAuthClient
-}
-
-// An ApplicationsConfig allows for the customization of an ApplicationsServer.
-// Use the DefaultConfig() to use the default dependencies.
-type ApplicationsConfig struct {
-	Logger           logr.Logger
-	Factory          services.Factory
-	JwtClient        auth.JWTClient
-	KubeClient       client.Client
-	GithubAuthClient auth.GithubAuthClient
-	Fetcher          applicationv2.Fetcher
-	GitlabAuthClient auth.GitlabAuthClient
-}
-
-// NewApplicationsServer creates a grpc Applications server
-func NewApplicationsServer(cfg *ApplicationsConfig) pb.ApplicationsServer {
-	return &applicationServer{
-		jwtClient:    cfg.JwtClient,
-		log:          cfg.Logger,
-		factory:      cfg.Factory,
-		kube:         cfg.KubeClient,
-		ghAuthClient: cfg.GithubAuthClient,
-		fetcher:      cfg.Fetcher,
-		glAuthClient: cfg.GitlabAuthClient,
-	}
-}
-
-// DefaultConfig creates a populated config with the dependencies for a Server
-func DefaultConfig() (*ApplicationsConfig, error) {
-	zapLog, err := zap.NewDevelopment()
-	if err != nil {
-		log.Fatalf("could not create zap logger: %v", err)
-	}
-
-	logr := zapr.NewLogger(zapLog)
-
-	rand.Seed(time.Now().UnixNano())
-	secretKey := rand.String(20)
-	jwtClient := auth.NewJwtClient(secretKey)
-
-	rest, clusterName, err := kube.RestConfig()
-	if err != nil {
-		return nil, fmt.Errorf("could not create client config: %w", err)
-	}
-
-	_, rawClient, err := kube.NewKubeHTTPClientWithConfig(rest, clusterName)
-	if err != nil {
-		return nil, fmt.Errorf("could not create kube http client: %w", err)
-	}
-
-	fluxClient := flux.New(osys.New(), &runner.CLIRunner{})
-
-	return &ApplicationsConfig{
-		Logger:           logr,
-		Factory:          services.NewServerFactory(fluxClient, internal.NewApiLogger(zapLog), nil, ""),
-		JwtClient:        jwtClient,
-		KubeClient:       rawClient,
-		Fetcher:          applicationv2.NewFetcher(rawClient),
-		GithubAuthClient: auth.NewGithubAuthClient(http.DefaultClient),
-		GitlabAuthClient: auth.NewGitlabAuthClient(http.DefaultClient),
-	}, nil
+type Config struct {
+	AppConfig      *ApplicationsConfig
+	ProfilesConfig ProfilesConfig
 }
 
-// NewApplicationsHandler allow for other applications to embed the Weave GitOps HTTP API.
-// This handler can be muxed with other services or used as a standalone service.
-func NewApplicationsHandler(ctx context.Context, cfg *ApplicationsConfig, opts ...runtime.ServeMuxOption) (http.Handler, error) {
-	appsSrv := NewApplicationsServer(cfg)
-
-	mux := runtime.NewServeMux(middleware.WithGrpcErrorLogging(cfg.Logger))
-	httpHandler := middleware.WithLogging(cfg.Logger, mux)
-	httpHandler = middleware.WithProviderToken(cfg.JwtClient, httpHandler, cfg.Logger)
+func NewApplicationsAndProfilesHandler(ctx context.Context, cfg *Config, opts ...runtime.ServeMuxOption) (http.Handler, error) {
+	mux := runtime.NewServeMux(middleware.WithGrpcErrorLogging(cfg.AppConfig.Logger))
```

Comment:
I had imagined apps and profiles being two separate muxes. Not sure if that is possible or not, however.

Reply:
Yeah I wasn't 100% sure the approach I went was the best. This approach means they share the same middle ware which is nice. Open to alternatives